datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -68,6 +70,8 @@ lazy_static! {
 pub struct FileScanConfig {
     /// Store from which the `files` should be fetched
     pub object_store: Arc<dyn ObjectStore>,
+    /// Object store URL
+    pub object_store_url: ObjectStoreUrl,
```

Comment:
This was the solution I came up with to allow ballista to round-trip the physical plan, store a canonical ObjectStoreUrl instead of relying on the files to contain the scheme.

 I think perhaps in a subsequent PR I'll try to remove the `object_store` member which is now redundant.

FYI @thinkharderdev

Reply:
worth a tracking ticket?